### PR TITLE
build(ui-dev): Fix API CSRF issues

### PR DESCRIPTION
**Pull request by billyvg**
This fixes API CSRF issues when developing on UI locally and proxying to production API.

Thanks @byk for the help!

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: webpack.config.js [hunk 1]
```diff
@@ -474,6 +474,9 @@ if (IS_UI_DEV_ONLY) {
         target: 'https://sentry.io',
         secure: false,
         changeOrigin: true,
+        headers: {
```
Comment (leedongwei) [inline webpack.config.js:477]: The comment in Line 460 should be updated because it is possible to write to the API now.

**Review by leedongwei — COMMENTED** [review]
This isn't within the scope of this PR, but just wanted to check if it'll make sense to replace `api.requestPromise` on this env and make it do nothing if the request method isn't `GET`?

**Comment by billyvg** [comment]
@leedongwei Do you mean, you would prefer that we didn't allow people to write to API?

**Review by matejminar — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]

**Comment by leedongwei** [comment]
@billyvg Allowing write requests to the API from the dev environment is dangerous and shouldn't be the default. 

I can see how it'll be useful for the design team to be able to go through flows (e.g. creating a Discover query etc) but IMO it'll be better to put it behind a flag so folks will understand the implications. We could draw inspiration from `dangerouslysetinnerhtml` in React, so something like `yarn dev-ui --dangerously-allow-writes`.

**Comment by billyvg** [comment]
@leedongwei I would completely agree if you could access our internal org, but you won't be able to because SSO is required. I think it's less dangerous given that you only have access to a test organization.

**Review by leedongwei — APPROVED** [review]
SGTM. I just wanted to be sure that we've considered the implications.

**Pull request merged**